### PR TITLE
Add Model Zoo testing support

### PR DESCRIPTION
**Pull request by attila-dusnoki-htec**
This change adds 2 "Model Zoo" to test:
- ONNX Zoo (https://onnx.ai/models/ and https://github.com/onnx/models)
  - The script will use the repository, and does automatically the following steps:
    - Downloads the models one-by-one with the provided test sample
    - Unzips and converts everything to test_runners.py format
    - Runs the test
    - Cleans up after the run
    - Repeats for all selected models
- Sample Generator with Datasets
  - The provided pyhon modul can download models and generated test samples from real datasets for them
  - It is extendable with furter datasets and models
  - Current support
    - 3 Datasets for image, text and audio
      - ImageNet 2012 Val, SQuAD v1.1, LibriSpeech ASR
    - Models using those datasets
      - ResNet50 v1, ResNet50 v1.5, TIMM MobileNetv3-large, ViT-base-patch16-224, CLIP-ViT-large-patch14
      - BERT-large-uncased, DistilBERT-base-cased-distilled, Roberta-base, GPT-J, Llama2-7b-chat-hf, T5-base, Gemma-2b-it
      - Wav2Vec2-base-960h, Whisper-small-en

### CODE REVIEW

File: tools/model_zoo/test_generator/README.md [hunk 1]
```diff
@@ -0,0 +1,113 @@
...
+python generate.py
+```
+
+To generate a subset of the supported models:
+- `none` to skip it
+- `all` for every models
+- <name> list supported model names
+
+```bash
+ptython --image resnet50_v1.5 clip-vit-large-patch14 --text none --audio none
```
Comment (attila-dusnoki-htec) [inline tools/model_zoo/test_generator/README.md:58]: `ptython --image resnet50_v1.5 clip-vit-large-patch14 --text none --audio none` ->
`python generate.py --image resnet50_v1.5 clip-vit-large-patch14 --text none --audio none`

File: tools/model_zoo/test_generator/requirements.txt [hunk 2]
```diff
@@ -0,0 +1,41 @@
...
+# THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+# IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+# FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT.  IN NO EVENT SHALL THE
+# AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+# LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+# OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+# THE SOFTWARE.
+#
+#####################################################################################
+-f https://repo.radeon.com/rocm/manylinux/rocm-rel-6.0/
```
Comment (kahmed10) [inline tools/model_zoo/test_generator/requirements.txt:25]: can you bump the version for this?

File: tools/model_zoo/onnx_zoo/README.md [hunk 3]
```diff
@@ -0,0 +1,48 @@
+# ONNX Zoo model tester
+
+Helper script to test [`ONNX Zoo models`](https://onnx.ai/models/) which have test data with [`test_runner.py`](../../test_runner.py)
+
+## Getting the repository
+
+*Important: make sure to enable git-lfs*
```
Comment (kahmed10) [inline tools/model_zoo/onnx_zoo/README.md:7]:
```suggestion
> [!IMPORTANT]
> Make sure to enable git-lfs.
```

File: tools/model_zoo/onnx_zoo/README.md [hunk 4]
```diff
@@ -0,0 +1,48 @@
...
+
+```bash
+git clone https://github.com/onnx/models.git --depth 1
+```
+
+## Running the tests
+
+*Important: the argument must point to a folder, not a file*
```
Comment (kahmed10) [inline tools/model_zoo/onnx_zoo/README.md:15]:
```suggestion
> [!IMPORTANT]
> The argument must point to a folder, not a file.
```

File: tools/model_zoo/test_generator/README.md [hunk 5]
```diff
@@ -0,0 +1,115 @@
...
+                        Audio models to test with librispeech-asr dataset samples
+  --output-folder-prefix OUTPUT_FOLDER_PREFIX
+                        Output path will be "<this-prefix>/<dataset-name>/<model-name>"
+  --sample-limit SAMPLE_LIMIT
+                        Max number of samples generated. Use 0 to ignore it.
+  --decode-limit DECODE_LIMIT
+                        Max number of sum-samples generated for decoder models. Use 0 to ignore it. (Only for decoder models)
+```
+
+Note: Some models require permission to access, use `huggingface-cli login`
```
Comment (kahmed10) [inline tools/model_zoo/test_generator/README.md:45]:
```suggestion
> [!NOTE]
> Some models require permission to access, use `huggingface-cli login`
```

File: tools/model_zoo/test_generator/README.md [hunk 6]
```diff
@@ -0,0 +1,115 @@
...
+- `model_id`: Hugging Face url
+- `name`: unique name for model
+
+To add a more complex model (e.g. Decoder), check [text](./sample_generator/model/text.py).
+
+The [generate](./generate.py) part will need further updating to include the model.
+
+## Adding more datasets
+
+The 3 most common usecase are handled:
```
Comment (kahmed10) [inline tools/model_zoo/test_generator/README.md:108]:
```suggestion
The 3 most common use cases are handled:
```

File: tools/model_zoo/test_generator/README.md [hunk 7]
```diff
@@ -0,0 +1,115 @@
...
+- `Image`:  with [imagenet](./sample_generator/dataset/imagenet.py)
+- `Text`:  with [squad](./sample_generator/dataset/squad.py)
+- `Audio`:  with [librispeech](./sample_generator/dataset/librispeech.py)
+
+To add a new, e.g. Video, create a new python file in dataset, and inherit a new class from Base.
```
Comment (kahmed10) [inline tools/model_zoo/test_generator/README.md:113]:
```suggestion
To add a new use case, e.g. Video, create a new python file in dataset, and inherit a new class from Base.
```

**Review by kahmed10 — APPROVED** [review]
From review side, looks good. Would need someone to test and verify it works. @causten

**Review by aarushjain29 — APPROVED** [review]
It is working without any errors. By default, the numpy 2.0 is there but it requires numpy 1.24.1

**Comment by attila-dusnoki-htec** [comment]
> It is working without any errors. By default, the numpy 2.0 is there but it requires numpy 1.24.1

Oh right, it was released not long ago.
Probably everything needs to be pinned to a version, not just the critical part like torch.

File: tools/model_zoo/test_generator/requirements.txt [hunk 8]
```diff
@@ -0,0 +1,41 @@
...
+# THE SOFTWARE.
+#
+#####################################################################################
+-f https://repo.radeon.com/rocm/manylinux/rocm-rel-6.1/
+torch == 2.1.2
+torchvision == 0.16.1
+optimum >= 1.19.0
+accelerate
+datasets
+numpy
```
Comment (causten) [inline tools/model_zoo/test_generator/requirements.txt:31]:
```suggestion
numpy == 1.24.1
```

**Pull request merged**